presto-main/src/test/java/com/facebook/presto/operator/scalar/AbstractTestFunctions.java

```diff
@@ -41,27 +43,50 @@
 import static com.facebook.presto.spi.StandardErrorCode.NUMERIC_VALUE_OUT_OF_RANGE;
 import static com.facebook.presto.spi.type.DecimalType.createDecimalType;
 import static com.facebook.presto.type.UnknownType.UNKNOWN;
+import static io.airlift.testing.Closeables.closeAllRuntimeException;
 import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
 import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.fail;
 
 public abstract class AbstractTestFunctions
 {
-    protected final FunctionAssertions functionAssertions;
+    private final Session session;
+    private final FeaturesConfig config;
+    protected FunctionAssertions functionAssertions;
 
     protected AbstractTestFunctions()
     {
-        functionAssertions = new FunctionAssertions();
+        this(TEST_SESSION);
     }
 
     protected AbstractTestFunctions(Session session)
     {
-        functionAssertions = new FunctionAssertions(session);
+        this(session, new FeaturesConfig());
     }
 
-    protected AbstractTestFunctions(FeaturesConfig featuresConfig)
+    protected AbstractTestFunctions(FeaturesConfig config)
     {
-        functionAssertions = new FunctionAssertions(TEST_SESSION, featuresConfig);
+        this(TEST_SESSION, config);
+    }
+
+    protected AbstractTestFunctions(Session session, FeaturesConfig config)
+    {
+        this.session = requireNonNull(session, "session is null");
+        this.config = requireNonNull(config, "config is null");
+    }
+
+    @BeforeClass
+    public void setUp()
+    {
+        functionAssertions = new FunctionAssertions(session, config);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        closeAllRuntimeException(functionAssertions);
```

Comment:
There's no need to use `closeAllRuntimeException` for closing a single object. The `@AfterClass` method can simply say `throws IOException`. That seems cleaner and more straight forward.

The method exists to allow closing multiple things and is arguably not needed since the same can be done trivially using try-with-resources. We should probably deprecate it.